openedx/core/djangoapps/waffle_utils/__init__.py

```diff
@@ -376,10 +376,11 @@ def is_enabled(self, course_key=None):
             course_key (CourseKey): The course to check for override before
             checking waffle.
         """
-        # validate arguments
-        assert issubclass(type(course_key), CourseKey), "The course_key '{}' must be a CourseKey.".format(
-            str(course_key)
-        )
+        if course_key is not None:
```

Comment:
@robrap here's the changes I'm making to the `CourseWaffleFlag` that we discussed yesterday.

Reply:
This is what I would propose to help people avoid bugs.  I don't want people to forget to pass in `course_key` and skip overrides.  Let me know your thoughts.

I'd add an argument like `skip_course_override=False`, where the user must do exactly one of the following:
1. Supply `course_key`, or
2. Set `skip_course_override=True`

Maybe the comment would be:
```
    skip_course_override (bool): Default is False, which uses `course_key` to check override.  If True, `course_key` must be None. This is useful for a course scoped flag that also must protect code where there is no course.
```

FYI: @nasthagiri

Reply:
Have you also considered having 2 separate public methods for this?  Separate interfaces for these 2 different behaviors will allow you to have methods with fewer optional arguments.  Also, if a self-documenting name is chosen for the 2 methods, there would be less reliance on the comment/docstring.

Reply:
I did and forgot and I’m glad I tagged you. I agree. Something like `is_enabled_no_course_override` or `is_enabled_no_override`? I don’t have a great name.